src/logic/calculate.js

```diff
@@ -0,0 +1,49 @@
+import operate from './operate';
+
+const calculate = (data, buttonName) => {
+  const { total, next, operation } = data;
+
+  let newTotal = total;
+  let newNext = next;
+  let newOperation = operation;
+
+  if (
+    buttonName === '+'
+    || buttonName === '-'
+    || buttonName === '×'
+    || buttonName === '÷'
+  ) {
+    newOperation = buttonName;
+    if (newNext) {
+      newTotal = operate(newTotal, newNext, newOperation);
+      newNext = null;
+      newOperation = null;
+    }
+  } else if (buttonName === '%') {
+    if (newNext) {
+      newTotal = operate(newNext, '%');
+      newNext = null;
+      newOperation = null;
+    } else if (newTotal) {
+      newTotal = operate(newTotal, '%');
+      newNext = null;
+      newOperation = null;
+    }
+  } else if (buttonName === '+/-') {
+    if (newNext) {
+      newTotal = operate(newNext, -1, '×');
+    } else if (newTotal) {
+      newTotal = operate(newTotal, -1, '×');
+    }
+    newNext = null;
+    newOperation = null;
+  } else if (buttonName === 'AC') {
+    newTotal = null;
+    newNext = null;
+    newOperation = null;
+  }
+
+  return { newTotal, newNext, newOperation };
+};
```

Comment:
**[OPTIONAL]** Since the scope of this review is not checking if the actual logic works, I'm leaving this as an optional change, lets take a look at your function if i call calculate with a button name of `+` and I have a newNext value of 5,  we are going to execute the operate function with just one value, because the newTotal has no value yet, so we will be calling it like so `operate(null, 5, +)` which will not make much sense. If you want you can revisit your logic onw, or you can do it on the next milestone where it is going to start making more sense, since you are going to have to connect your logic to your UI.